docs/dialects.dot

```diff
@@ -0,0 +1,123 @@
+    Standard [fillcolor="lightblue"]
+    Arith [fillcolor="lightblue"]
+    Affine [fillcolor="lightblue"]
+  }
+
+  // ===== Nodes =====
+
+  subgraph cluster_circt {
+    label = "CIRCT"
+    // CIRCT dialects
+    Calyx [URL="https://circt.llvm.org/docs/Dialects/Calyx/"]
+    Handshake [URL="https://circt.llvm.org/docs/Dialects/Handshake/"]
+    FIRRTL [URL="https://circt.llvm.org/docs/Dialects/FIRRTL/"]
+    LLHD [URL="https://circt.llvm.org/docs/Dialects/LLHD/"]
+    HW [URL="https://circt.llvm.org/docs/Dialects/HW/"]
+    SV [URL="https://circt.llvm.org/docs/Dialects/SV/"]
+    Comb [URL="https://circt.llvm.org/docs/Dialects/Comb/"]
+    Seq [URL="https://circt.llvm.org/docs/Dialects/Seq/"]
+    StaticLogic [URL="https://circt.llvm.org/docs/Dialects/StaticLogic/"]
+    MSFT [URL="https://circt.llvm.org/docs/Dialects/MSFT/"]
+    ESI [URL="https://circt.llvm.org/docs/Dialects/ESI/"]
+    FSM [URL="https://circt.llvm.org/docs/Dialects/FSM/"]
+
+    // Internal tools
+    ExportVerilog [fillcolor="darksalmon"]
+    FIRRTLParser [label="FIRRTL Parser" fillcolor="darksalmon"]
+    Scheduling [fillcolor="darksalmon"]
+    llhd_sim [label="LLHD Sim" fillcolor="darksalmon"]
+    PyCDE [fillcolor="darksalmon"]
+  }
+
+  // External tools
+  Moore [shape=octagon fillcolor="gold"]
+  Circilator [shape=octagon fillcolor="gold" style="dashed,filled"]
+  Calyx_native [label="Calyx Native" shape=octagon fillcolor="gold"]
+
+  // File formats
+  VCDTrace [label=".vcd" fillcolor="yellowgreen" shape=note]
+  CPPFile [label=".cpp" fillcolor="yellowgreen" shape=note]
+  SVFile [label=".sv" fillcolor="yellowgreen" shape=note]
+  CapNProto [label="Cap'n Proto" fillcolor="yellowgreen" shape=note]
+  TCL [label=".tcl" fillcolor="yellowgreen" shape=note]
+  FIRFile [label=".fir" fillcolor="yellowgreen" shape=note]
+  SVVHDL [label=".sv/.vhd" fillcolor="yellowgreen" shape=note]
+  PyFile [label=".py" fillcolor="yellowgreen" shape=note]
+
+  // ===== Connections =====
+
+  // Dialect conversions
+  SCF -> Calyx
+  Affine -> StaticLogic [style=dashed]
+  StaticLogic -> Calyx [style=dashed]
+  Standard -> Calyx
+  Arith -> Calyx
+  Standard -> Handshake
+  Arith -> Handshake
+
+  Handshake -> FIRRTL
+  Calyx -> FSM [style=dashed]
+  Seq -> SV
+  FIRRTL -> HW
+  FSM -> HW
+  ESI -> HW
+  MSFT -> HW
+  HW -> LLHD
+  FIRRTL->Comb
+  Comb -> ExportVerilog
+
+  // Tool flows
+  Scheduling -> StaticLogic [dir=both]
+  FIRFile -> FIRRTLParser
+  FIRRTLParser -> FIRRTL
+  HW -> ExportVerilog
+  SV -> ExportVerilog
+  SVVHDL -> Moore [weight=999]
+  Moore -> LLHD
+  LLHD -> Circilator [style=dashed]
+  HW -> Circilator [style=dashed]
+  Calyx -> Calyx_native
+  ExportVerilog -> SVFile
+  Circilator -> CPPFile [style=dashed]
+  LLHD -> llhd_sim
+  Comb -> llhd_sim
+  HW -> llhd_sim
+  Seq -> llhd_sim
+  llhd_sim -> VCDTrace
+  Calyx_native -> SVFile
+  ESI -> CapNProto
+  MSFT -> TCL
+  PyFile -> PyCDE
+  PyCDE -> HW
+  PyCDE -> Comb
+  PyCDE -> Seq
+  PyCDE -> LLHD
```

Comment:
PyCDE doesn't directly produce LLHD... though it'd be interesting to expose some of LLHD's timing constructs through PyCDE.

Reply:
Thanks for adding it!

Reply:
Would you prefer a dotted edge between PyCDE and LLHD, or no edge? I guess there's also a danger in adding hypothetical edges all over, cluttering up the graph :thinking:.